src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2411,15 +2453,19 @@ private static ImmutableArray<Symbol> FixPartialMember(ImmutableArray<Symbol> sy
             SourceMemberMethodSymbol.InitializePartialMethodParts(definition, implementation);
 
             // a partial method is represented in the member list by its definition part:
+            return Remove(symbols, implementation);
+        }
+
+        private static ImmutableArray<Symbol> Remove(ImmutableArray<Symbol> symbols, Symbol symbol)
```

Comment:
Added a `PROTOTYPE` comment. Note this would only allow removing all replace symbols at once that have the same name.